core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -66,6 +66,7 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
       host: String,
       port: Int,
       execId: String,
+      shuffleGenerationId: Int,
```

Comment:
does `BlockTransferService` only handle shuffle blocks?

Reply:
IIUC, fetchBlockSync/uploadBlockSync handle non-shuffle blocks.

Reply:
then it's unnecessary to have a `shuffleGenerationId` parameter when fetching non-shuffle blocks?

Reply:
Yep.

Reply:
shall we create a `fetchShuffleBlocks` method?

Reply:
This `fetchBlocks` override from `ShuffleClient`, and it only used fetching non-shuffle blocks in `fetchBlockSync`, so maybe we add a comment in fetchBlockSync and point this out?

Reply:
Add more comment in fetchBlockSync in 2362512.